### DISCUSSION TITLE
BuyerOrderReferencedDocument fails validation: Document MUST not contain empty elements

### DISCUSSION THREAD

**Discussion by cdrfun** [stephanstapel/ZUGFeRD-csharp#468] [General]
Hi there,
I'm trying to generate a valid XRechnung, but I think there might be a mistake when applying BuyerOrderReferencedDocument information.

I've used the Method `AddTradeLineItem` to set buyerOrderID and buyerOrderDate:
```
TradeLineItem.AddTradeLineItem([...], buyerOrderID: <OrderNo>, buyerOrderDate: OrderDate, [...])
```

When validating the resulting using mustang:
```
java -Xmx1G -jar Mustang-CLI-2.15.0.jar --action validate --source "zugferd-test.xml"
```

I get the following error:
```
<warning type="27" location="/rsm:CrossIndustryInvoice/rsm:SupplyChainTradeTransaction[1]/ram:IncludedSupplyChainTradeLineItem[1]/ram:SpecifiedLineTradeAgreement[1]/ram:BuyerOrderReferencedDocument[1]" criterion="false()">Document MUST not contain empty elements. [ID PEPPOL-EN16931-R008] from /xslt/XR_30/XRechnung-CII-validation.xslt)</warning>
```

After trying to hunt down the problem I stumbled accross this section: https://github.com/stephanstapel/ZUGFeRD-csharp/blob/863d58a23a925ebfd63bfc1b65af77cef01d5003/ZUGFeRD/InvoiceDescriptor23CIIWriter.cs#L239-L263

My applied profile is `Profile.XRechnung` and I'm using `ZUGFeRDVersion.Version23`.

I'm not totally shure, but I think the issue is, that in line 246 `Writer.WriteOptionalElementString("ram", "IssuerAssignedID", tradeLineItem.BuyerOrderReferencedDocument.ID, Profile.Extended);` the profile is set to `Extended`. From my understanding it must be equal to line 243 (`Profile.Comfort | Profile.Extended | Profile.XRechnung1 | Profile.XRechnung`).

If someone could verify my findings I'd be happy to make a pull request.

**Comment by m4Viper** [comment]
yes, I have the same error with XRechnung and BuyerOrderReferencedDocument

**Comment by stephanstapel** [comment]
@cdrfun : please go ahead!
Can you please also check for ZUGFeRD 2.0 and 1.x?

**Comment by tobitege** [comment]
This is now fixed with the merge of my PR #549
